ast/compile.go

```diff
@@ -4761,7 +4761,12 @@ func checkUnusedAssignedVars(loc *Location, stack *localDeclaredVars, used VarSe
 	unused = unused.Diff(rewrittenUsed)
 
 	for _, gv := range unused.Sorted() {
-		errs = append(errs, NewError(CompileErr, loc, "assigned var %v unused", dvs.reverse[gv]))
+		for i := range body {
+			if body[i].Vars(VarVisitorParams{}).Contains(gv) {
+				// TODO: better way to get location of assignment?
+				errs = append(errs, NewError(CompileErr, body[i].Loc(), "assigned var %v unused", dvs.reverse[gv]))
+			}
```

Comment:
In `checkUnusedDeclaredVars()`, we have a special case for when the var couldn't be found in the body. From the top of my head, I don't know what might trigger that case; but just to be safe, we should probably do the same here, as on a miss, we'd drop the error and nothing would be reported.

Reply:
I tried putting the following
```go
    bodyvars := cpy.Vars(VarVisitorParams{})
    for v := range used {
        if gv, ok := stack.Declared(v); ok {
            bodyvars.Add(gv)
        } else {
            bodyvars.Add(v)
        }
    }
    unused = unused.Diff(bodyvars).Diff(used)
```
but seems with this form, it causes a regression, as it can’t catch the cases for `assigned var x unused`, only arg vars are being caught
Any idea what is missing? (imo since we iterate over rewrittenUsed
, there’s no need to check the body parameters for the second time)

Reply:
I think if the var couldn't be found in the body, we should just use the location of the first line of the body. This scenario shouldn't happen, I think, but it's better to hedge our bets here rather than dropping an error.